### PR TITLE
Fix NRE in ConsiceView

### PR DESCRIPTION
**Pull request by iSazonov**
# PR Summary

Use LanguagePrimitives.TryConvertTo() as last resort to string conversion.

## PR Context

Playing with AccountManagement module I catch an error where properties with disposed object reference present.
```powershell
PS C:\Windows\System32> $e.psobject.properties

GetterScript    : & { Set-StrictMode -Version 1; $this.Exception.InnerException.PSMessageDetails }
SetterScript    :
MemberType      : ScriptProperty
IsSettable      : False
IsGettable      : True
Value           :
TypeNameOfValue : System.Object
Name            : PSMessageDetails
IsInstance      : False

MemberType      : Property
Value           : System.Management.Automation.MethodInvocationException: Exception calling "ToString" with "0"
                  argument(s): "Can�t delete an already deleted object"
                   ---> System.InvalidOperationException: Can�t delete an already deleted object
                     at System.DirectoryServices.AccountManagement.Principal.CheckDisposedOrDeleted()
                     at System.DirectoryServices.AccountManagement.Principal.HandleGet[T](T& currentValue, String
                  name, LoadState& state)
                     at System.DirectoryServices.AccountManagement.Principal.get_Name()
                     at System.DirectoryServices.AccountManagement.Principal.ToString()
                     at CallSite.Target(Closure , CallSite , Object )
                     --- End of inner exception stack trace ---
                     at System.Management.Automation.ExceptionHandlingOps.ConvertToMethodInvocationException(Exception
                  exception, Type typeToThrow, String methodName, Int32 numArgs, MemberInfo memberInfo) in C:\Users\sie
                  \Documents\GitHub\iSazonov\PowerShell\src\System.Management.Automation\engine\runtime\Operations\Misc
                  Ops.cs:line 2062
                     at CallSite.Target(Closure , CallSite , Object )
                     at System.Dynamic.UpdateDelegates.UpdateAndExecute1[T0,TRet](CallSite site, T0 arg0)
                     at CallSite.Target(Closure , CallSite , Object )
                     at System.Management.Automation.Interpreter.DynamicInstruction`2.Run(InterpretedFrame frame) in C:
                  \Users\sie\Documents\GitHub\iSazonov\PowerShell\src\System.Management.Automation\engine\interpreter\D
                  ynamicInstructions.Generated.cs:line 141
                     at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame
                  frame) in C:\Users\sie\Documents\GitHub\iSazonov\PowerShell\src\System.Management.Automation\engine\i
                  nterpreter\ControlFlowInstructions.cs:line 357
IsSettable      : False
IsGettable      : True
TypeNameOfValue : System.Exception
Name            : Exception
IsInstance      : True

MemberType      : Property
Value           :
IsSettable      : False
IsGettable      : True
TypeNameOfValue : System.Object
Name            : TargetObject
IsInstance      : True

```

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [ ] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

**Review by SteveL-MSFT — APPROVED** [review]
Possible to add a test?  Seems like we can create an ErrorRecord that hits this code path?

**Comment by iSazonov** [comment]
> Possible to add a test? Seems like we can create an ErrorRecord that hits this code path?

@SteveL-MSFT I did not find how to do it :-(

**Comment by SteveL-MSFT** [comment]
@iSazonov seems like line 802 would have already covered this?  Can you show how it looks before and after this change with your error?

**Comment by adityapatwardhan** [comment]
@iSazonov Can you provide a repro with your module?

**Comment by iSazonov** [comment]
My fix is not right.

> $value = $prop.Value.ToString().Trim()

The line throws if:
- the $prop.Value contains a reference to disposed object
- ToString() throws

I will update the PR.

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]
Ok, so it's calling a method on a disposed object.

File: src/System.Management.Automation/FormatAndOutput/DefaultFormatters/PowerShellCore_format_ps1xml.cs [hunk 1]
```diff
@@ -883,25 +883,28 @@ private static IEnumerable<FormatViewDefinition> ViewsOf_System_Management_Autom
                                                 }
```
Comment (PoshChan) [inline src/System.Management.Automation/FormatAndOutput/DefaultFormatters/PowerShellCore_format_ps1xml.cs]: @iSazonov, your last commit had 3 failures in `PowerShell-CI-windows`
Get-Error tests.Get-Error will accept pipeline input
```powershell
Expected case sensitive like wildcard '*-2146233087*' to match '
Exception             : 
    Type           : 
    ErrorRecord    : 
        Exception             : 
            Type    : 
            Message : 
            HResult : 
        CategoryInfo          : 
        FullyQualifiedErrorId : 
        InvocationInfo        : 
            ScriptLineNumber : 
            OffsetInLine     : 
            HistoryId        : 
            ScriptName       : 
            Line             : 
            PositionMessage  : 
            PSScriptRoot     : 
            PSCommandPath    : 
            CommandOrigin    : 
        ScriptStackTrace      : 
    TargetSite     : 
        Name          : 
        DeclaringType : 
        MemberType    : 
        Module        : 
    StackTrace     : 
   at System.Management.Automation.ExceptionHandlingOps.CheckActionPreference(FunctionContext funcContext, Exception exception)
   at System.Management.Automation.Interpreter.ActionCallInstruction`2.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
    Message        : 
    Data           : 
    InnerException : 
        Type    : 
        Message : 
        HResult : 
    Source         : 
    HResult        : 
CategoryInfo          : 
FullyQualifiedErrorId : 
InvocationInfo        : 
    ScriptLineNumber : 
    OffsetInLine     : 
    HistoryId        : 
    ScriptName       : 
    Line             : 
    PositionMessage  : 
    PSScriptRoot     : 
    PSCommandPath    : 
    CommandOrigin    : 
ScriptStackTrace      : 

', but it did not match.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Utility\Get-Error.Tests.ps1: line 75
75:         $out | Should -BeLikeExactly '*-2146233087*'
```
Get-Error tests.Get-Error will handle Exceptions
```powershell
Expected case sensitive like wildcard '*myexception*' to match '
Type    : 
Message : 
HResult : 

', but it did not match.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Utility\Get-Error.Tests.ps1: line 83
83:         $out | Should -BeLikeExactly '*myexception*'
```
Get-Error tests.Get-Error adds ExceptionType for Exceptions
```powershell
Expected case sensitive like wildcard '*System.Management.Automation.ParentContainsErrorRecordException*' to match '
Exception             : 
    Type           : 
    ErrorRecord    : 
        Exception             : 
            Type    : 
            Message : 
            HResult : 
        CategoryInfo          : 
        FullyQualifiedErrorId : 
        InvocationInfo        : 
            ScriptLineNumber : 
            OffsetInLine     : 
            HistoryId        : 
            ScriptName       : 
            Line             : 
            PositionMessage  : 
            PSScriptRoot     : 
            PSCommandPath    : 
            CommandOrigin    : 
        ScriptStackTrace      : 
    TargetSite     : 
        Name          : 
        DeclaringType : 
        MemberType    : 
        Module        : 
    StackTrace     : 
   at System.Management.Automation.ExceptionHandlingOps.CheckActionPreference(FunctionContext funcContext, Exception exception)
   at System.Management.Automation.Interpreter.ActionCallInstruction`2.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
    Message        : 
    Data           : 
    InnerException : 
        Type            : 
        Message         : 
        SocketErrorCode : 
        ErrorCode       : 
        NativeErrorCode : 
        TargetSite      : 
            Name          : 
            DeclaringType : 
            MemberType    : 
            Module        : 
        StackTrace      : 
   at System.Net.Dns.InternalGetHostByName(String hostName)
   at System.Net.Dns.GetHostByName(String hostName)
   at CallSite.Target(Closure , CallSite , Type , Object )
        Source          : 
        HResult         : 
    Source         : 
    HResult        : 
CategoryInfo          : 
FullyQualifiedErrorId : 
InvocationInfo        : 
    ScriptLineNumber : 
    OffsetInLine     : 
    HistoryId        : 
    ScriptName       : 
    Line             : 
    PositionMessage  : 
    PSScriptRoot     : 
    PSCommandPath    : 
    CommandOrigin    : 
ScriptStackTrace      : 

', but it did not match.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Utility\Get-Error.Tests.ps1: line 126
126:         $out | Should -BeLikeExactly "*$expectedExceptionType*"
```

**Review by SteveL-MSFT — APPROVED** [review]

**Comment by SteveL-MSFT** [comment]
Tried to simulate the original issue of a disposed object, but couldn't find a way to do it.

**Review by adityapatwardhan — APPROVED** [review]

**Comment by iSazonov** [comment]
> Tried to simulate the original issue of a disposed object, but couldn't find a way to do it.

I too. I could port a code from AccountManagement module but I think it is not correct behavior for ToString() to throw and I will report the issue to Core team.

**Pull request merged**